### PR TITLE
docs: add label/postings offset tables

### PR DESCRIPTION
**Pull request by zhulongcheng**
Without reading the code, it is a little difficult to understand how the offset table work.

Add label/postings offset entries to make it clear:
```
label offset entry: label name -> label values
postings offset entry: label name/value -> series list
```

### CODE REVIEW

**Comment by zhulongcheng** [comment]
ping @krasi-georgiev

**Comment by krasi-georgiev** [comment]
maybe also update the links to take to each relevant section

`The sequence of label index sections is finalized by an offset table pointing ...`
and
`The sequence of postings sections is finalized by an offset table pointing ...`

by the way since you are poking around might also help update the tsdb doc to help other contributors get up to speed quickly. I usually use as a draft when looking around the code and trying to make sense of it all :smile: 

https://docs.google.com/document/d/1t5eSLlH36oK1Ex3PWwuc4R58kx4PC7Ago90TRR_7nmo

ping me if you want access and I will add you.

**Comment by zhulongcheng** [comment]
Wow, great doc!

**Comment by krasi-georgiev** [comment]
LGTM ping @codesome. @gouthamve

File: docs/format/index.md [hunk 1]
```diff
@@ -175,7 +175,7 @@ Postings sections store monotonically increasing lists of series references that
 └─────────────────────────────────────────┘
 ```
 
-The sequence of postings sections is finalized by an [offset table](#offset-table) pointing to the beginning of each postings section for a given set of label names.
+The sequence of postings sections is finalized by an [offset table](#posting-offset-entry) pointing to the beginning of each postings section for a given set of label names.
```
Comment (bwplotka) [inline docs/format/index.md]: This is quite confusing to me. The section is finalized by an Offset table itself NOT the posting / label one. So the full sentence should look like:
```suggestion
The sequence of postings sections is finalized by an [offset table](#offset-table) containing [posting offset entries](#posting-offset-entry) that points to the beginning of each postings section for a given set of label names.
```
and same for labels, right?

File: docs/format/index.md [hunk 2]
```diff
@@ -175,7 +175,7 @@ Postings sections store monotonically increasing lists of series references that
...
 
 ### Offset Table
```
Comment (bwplotka) [inline docs/format/index.md]: line below: Let's add that entries of this offset tables can be different depending on series/posting (introduce to further sections `### Posting Offset Entry` and `### Label Offset Entry`

File: docs/format/index.md [hunk 3]
```diff
@@ -153,7 +153,7 @@ For instance, a single label name with 4 different values will be encoded as:
 └────┴───┴───┴──────────────┴──────────────┴──────────────┴──────────────┴───────┘
 ```
 
-The sequence of label index sections is finalized by an [offset table](#offset-table) pointing to the beginning of each label index section for a given set of label names.
+The sequence of label index sections is finalized by an [offset table](#label-offset-entry) pointing to the beginning of each label index section for a given set of label names.
```
Comment (bwplotka) [inline docs/format/index.md]: same as my comment for postings.

File: docs/format/index.md [hunk 4]
```diff
@@ -202,6 +202,32 @@ An offset table stores a sequence of entries that maps a list of strings to an o
 └────────────────────────────────────────────┘
 ```
 
+<a name="label-offset-entry"></a>Every label entry holds the label name and the offset to its values in the label index section.
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply docs/format/index.md]: essentially `header` for markdown creates link.

[the remark this answers is not in the document]
Comment (krasi-georgiev) [reply docs/format/index.md]: yeah I thought the same , but it makes it bold and a bit hard on the eyes.
I don't mind either way.

File: docs/format/index.md (hunk 1 above)
Comment (krasi-georgiev) [reply docs/format/index.md]: yep that will do as well.

**Comment by zhulongcheng** [comment]
Thanks @bwplotka @krasi-georgiev 

Now add `Label Offset Table` and `Postings Offset Table`.

**Comment by krasi-georgiev** [comment]
Thanks!

**Pull request merged**